packages/help-center/src/components/help-center-article.tsx

```diff
@@ -53,26 +54,54 @@ export const HelpCenterArticle = () => {
 		}
 	}, [ post, query, sectionName ] );
 
+	useEffect( () => {
+		if ( error ) {
+			const tracksData = {
+				force_site_id: true,
+				location: 'help-center',
+				section: sectionName,
+				post_url: postUrl,
+			};
+
+			recordTracksEvent( 'calypso_helpcenter_article_api_error', tracksData );
+		}
+	}, [ postUrl, error, sectionName ] );
```

Comment:
It would be great if we can pass this as a callback to the query [here](https://github.com/Automattic/wp-calypso/blob/b11ea104748f4f6b4e06ba3214e3f37abfdb5435/packages/help-center/src/hooks/use-post-by-url.ts#L20) instead. So every user of this hook benefits, not only this page.

```js
onError() {
	const tracksData = {
				force_site_id: true,
				location: 'help-center',
				section: sectionName,
				post_url: postUrl,
			};

	recordTracksEvent( 'calypso_helpcenter_article_api_error', tracksData );
}
```

You can get section name like this

```js
const { sectionName } = useHelpCenterContext();
```

Reply:
Updated [here](https://github.com/Automattic/wp-calypso/tree/a156a8d7b935c6af223f0c936830515569a319c7), I asked you for a new review @alshakero